pkg/sql/pgwire/pgwirebase/encoding.go

```diff
@@ -38,7 +39,13 @@ import (
 	"github.com/lib/pq/oid"
 )
 
-const maxMessageSize = 1 << 24
+// maxMessageSize is the maximum size of any varlen string we will accept in the
+// wire protocol. Postgres has a limit of maxint32 on this quantity, but they
+// also have the freedom to not completely die on failures to allocate, so we're
+// a little more conservative.
+// 128 megabytes is twice our range size - exceeding this would cause other
```

Comment:
I think the note here about range size is misleading, or maybe I don't understand it well. You could have a SELECT with a bunch of parameters that could also exceed this amount, which wouldn't effect the range size too much. Similarly an INSERT with a bunch of data is fine as long as there are many rows. The ranges will split themselves if they are too large. This is only a problem if there's a single row that large. I'm just worried this comment will lead to wrong conclusions in the future about why we can't increase this limit.

Reply:
Oh, yeah, good point. I'll just delete the range size thing, it was flimsy justification anyway.